File: CS/cs/Service/XpoGate.svc.cs
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using System.ServiceModel;
using System.Configuration;
using DevExpress.Xpo.DB.Helpers;
using DevExpress.Xpo.DB.Exceptions;
using System;

namespace DXSample.Service {
    public class XpoGate :IXpoGate {
        static ICacheToCacheCommunicationCore dataStore;

        static XpoGate() {
            string conn = ConfigurationManager.ConnectionStrings["XpoConnection"].ConnectionString;
            dataStore = new DataCacheRoot(XpoDefault.GetConnectionProvider(conn, AutoCreateOption.None));
        }
        
        #region IXpoGate Members

        DataCacheModificationResult IXpoGate.ModifyData(DataCacheCookie cookie, ModificationStatement[] dmlStatements) {
            try {
                return dataStore.ModifyData(cookie, dmlStatements);
            } catch (LockingException ex) {
                throw new FaultException<LockingException>(ex);
            }
        }

        DataCacheResult IXpoGate.NotifyDirtyTables(DataCacheCookie cookie, params string[] dirtyTablesNames) {
            return dataStore.NotifyDirtyTables(cookie, dirtyTablesNames);
        }

        DataCacheResult IXpoGate.ProcessCookie(DataCacheCookie cookie) {
            return dataStore.ProcessCookie(cookie);
        }

        DataCacheSelectDataResult IXpoGate.SelectData(DataCacheCookie cookie, SelectStatement[] selects) {
            return dataStore.SelectData(cookie, selects);
        }

        #endregion
    }
}

File: VB/Service/XpoGate.svc.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports System.ServiceModel
Imports System.Configuration
Imports DevExpress.Xpo.DB.Helpers
Imports DevExpress.Xpo.DB.Exceptions
Imports System

Namespace DXSample.Service
	Public Class XpoGate
		Implements IXpoGate
		Private Shared dataStore As ICacheToCacheCommunicationCore

		Shared Sub New()
			Dim conn As String = ConfigurationManager.ConnectionStrings("XpoConnection").ConnectionString
			dataStore = New DataCacheRoot(XpoDefault.GetConnectionProvider(conn, AutoCreateOption.None))
		End Sub

		#Region "IXpoGate Members"

		Private Function ModifyData(ByVal cookie As DataCacheCookie, ByVal dmlStatements() As ModificationStatement) As DataCacheModificationResult Implements IXpoGate.ModifyData
			Try
				Return dataStore.ModifyData(cookie, dmlStatements)
			Catch ex As LockingException
				Throw New FaultException(Of LockingException)(ex)
			End Try
		End Function

		Private Function NotifyDirtyTables(ByVal cookie As DataCacheCookie, ParamArray ByVal dirtyTablesNames() As String) As DataCacheResult Implements IXpoGate.NotifyDirtyTables
			Return dataStore.NotifyDirtyTables(cookie, dirtyTablesNames)
		End Function

		Private Function ProcessCookie(ByVal cookie As DataCacheCookie) As DataCacheResult Implements IXpoGate.ProcessCookie
			Return dataStore.ProcessCookie(cookie)
		End Function

		Private Function IXpoGate_SelectData(ByVal cookie As DataCacheCookie, ByVal selects() As SelectStatement) As DataCacheSelectDataResult Implements IXpoGate.SelectData
			Return dataStore.SelectData(cookie, selects)
		End Function

		#End Region
	End Class
End Namespace
